client/src/components/DecisionButtons.vue

```diff
@@ -16,13 +21,45 @@ export default {
     return {
       warnDecision: false,
       newComment: '',
+      confirmedPresent: [],
+      confirmedAbsent: [],
     };
   },
   computed: {
+    ...mapState([
+      'currentViewData',
+    ]),
     ...mapGetters([
       'currentViewData',
       'myCurrentProjectRoles',
     ]),
+    artifacts() {
+      return this.MIQAConfig.artifact_options.map((name) => ({
+        value: name,
+        labelText: this.convertValueToLabel(name),
+      }));
+    },
+    chips() {
+      return this.artifacts.map((artifact) => [artifact, this.getCurrentChipState(artifact)]);
+    },
+    suggestedArtifacts() {
+      if (this.currentViewData.scanDecisions.length > 0) {
+        const lastDecisionArtifacts = _.last(_.sortBy(
+          this.currentViewData.scanDecisions, (dec) => dec.created,
+        )).user_identified_artifacts;
+        return Object.entries(lastDecisionArtifacts).filter(
+          ([, present]) => present === 1,
+        ).map(([artifactName]) => artifactName);
+      } if (this.currentViewData.currentAutoEvaluation) {
+        const predictedArtifacts = this.currentViewData.currentAutoEvaluation.results;
+        return Object.entries(predictedArtifacts).filter(
+          ([artifactName, percentCertainty]) => artifactName !== 'overall_quality'
+            && artifactName !== 'normal_variants'
+            && percentCertainty < this.MIQAConfig.auto_artifact_threshold,
+        ).map(([artifactName]) => artifactName.replace('no_', '').replace('full', 'partial'));
```

Comment:
I have absolutely no idea what this magic is. Can you add a little documentation for posterity?

Reply:
Added explanation comments.